### ISSUE TITLE
Escaping characters in json

### ISSUE THREAD

**Issue by Ndrou** [orientechnologies/orientdb#3449]
Hi !

I have a little problem when i try to update a record using the JSON syntax with **OrientDB 2.0 final** : 
```sql
UPDATE User SET name = "foo", prop = { "foo" : "return \"the bar is broken !\"" } WHERE name = "bar"
```
When i execute this query, the updated record looks like : 
```js
"prop": {
    "foo": "return \\\"the bar is broken !\\\""
}
```

This result is not correct.

Have a good day !

**Comment by seeden** [comment]
@luigidellaquila any progress?

**Comment by tdesc** [comment]
Hello,
We have same problem with OrientDB version : 2.1.7 and on version 2.2-SNAPSHOT.
on 2.1.7:
next query gives us java.lang.IllegalStateException: Missing closed string character: '"', position: 66

begin
let update_0 = update #27:592 set test="\n<p>Новый ра<span style=\"color: #99cc00;\">зд</span>ел ewrwed</p>\n<p>&nbsp;</p>\n<p>werwersdfsdfasdf</p>\n<p>werwer dddsdf</p>\n<p>&nbsp;</p>\n<p>&nbsp;</p>\n<p>wer<span style=\"color: #ff0000;\">wer</span></p>\n<p>w<span style=\"color: #99cc00;\">e</span></p>\n<p>rwe</p>\n<p>rwer</p>\n<p>&nbsp;</p>\n<p>&nbsp;</p>\n"
commit retry 100

while same query without transaction work fine.

update #27:592 set test="\n<p>Новый ра<span style=\"color: #99cc00;\">зд</span>ел ewrwed</p>\n<p>&nbsp;</p>\n<p>werwersdfsdfasdf</p>\n<p>werwer dddsdf</p>\n<p>&nbsp;</p>\n<p>&nbsp;</p>\n<p>wer<span style=\"color: #ff0000;\">wer</span></p>\n<p>w<span style=\"color: #99cc00;\">e</span></p>\n<p>rwe</p>\n<p>rwer</p>\n<p>&nbsp;</p>\n<p>&nbsp;</p>\n"

on 2.2 error is different:
com.orientechnologies.orient.core.sql.parser.TokenMgrError: Lexical error at line 1, column 65. Encountered: <EOF> after : "\"\\n<p>\u041d\u043e\u0432\u044b\u0439 \u0440\u0430<span style=\\\"color: #99cc00"

**Comment by seeden** [comment]
This issue is really annoying

**Comment by luigidellaquila** [comment]
Hi guys,

I'm working on this in 2.2, it's not fixed yet unfortunately

Luigi

**Comment by seeden** [comment]
Thank you @luigidellaquila

**Comment by luigidellaquila** [comment]
Hi guys,

This issue is fixed in 2.2.
I'm closing this ticket, please reopen if you see any other related problems

Thanks

Luigi
